features/support/shared_contexts/simple_git_repository.rb

```diff
@@ -34,13 +31,19 @@
     git.add(all: true)
     git.commit('First commit')
 
-    raise "Can't generate map" unless system("cd #{root} && rspec spec > /dev/null") # Generate crystalball map
+    raise "Can't generate map" unless generate_map
   end
 
   after do
     root.rmtree
   end
 
+  def generate_map
+    config = map_generator_config.to_s.chomp
+    replace(spec_helper, /# MAP_GENERATOR_CONFIG/, config)
```

Comment:
here you change `spec_helper` and generate a map
let's split this logic
it is better to change `spec_helper` and commit it separately. so, main logic of a feature will be executed on a clean repo without unexpected side-effects